### PR TITLE
fixes #511 to add a withSelector(LabelSelector) method on the DSL

### LINKED ISSUE #511: should we have a pods().withSelector(LabelSelector)
we have various withLabel() methods like `withLabels(Map<String,String>)`, `withLabel(String, String)` but should we have a withSelector(LabelSelector) which then wraps up iterating through the LabelSelector turning it into matchLabels etc?

### PR DESCRIPTION
**Pull request by jstrachan**

### CODE REVIEW

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/Filterable.java [hunk 1]
```diff
@@ -38,5 +42,7 @@
   T withFields(Map<String, String> labels);
 
   T withField(String key, String value);
+
+  T withSelector(LabelSelector selector);
```
Comment (jimmidyson) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/Filterable.java]: Can we call it `withLabelSelector` please, just to be consistent with other method names?

Comment (jstrachan) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/Filterable.java]: sure

**Comment by jimmidyson** [comment]
Thanks! Will merge on green :)

**Comment by jimmidyson** [comment]
[merge]

**Pull request merged**